### DISCUSSION TITLE
Nodelist.forEach

### DISCUSSION THREAD

**Discussion by naweiss** [babel/babel#13119] [Q&A] [answered]
I have a simple code in file named `index.js`:
```js
document.querySelectorAll('Anything').forEach(x => {
	console.log(x);
});
```

My .babelrc file contains:
```js
{
    "presets": [
        [
            "@babel/preset-env",
            {
                "targets": { "chrome": CHROME_VERSION }, 
                "useBuiltIns": "usage",
                "corejs": 3
            }
        ]
    ]
}
```

And I installed the following versin of packages:
- core-js: ^3.10.1
- @babel/cli: ^7.13.14
- @babel/core: ^7.13.14
- @babel/preset-env: ^7.13.12

The result of running `npx babel index.js` with CHROME_VERSION of 57 is:
```
"use strict";

require("core-js/modules/web.dom-collections.for-each.js");

document.querySelectorAll('Anything').forEach(x => {
  console.log(x);
});
```
And for CHROME_VERSION of 58 is:
```js
"use strict";

document.querySelectorAll('Anything').forEach(x => {
  console.log(x);
});
```

This implies that NodeList.forEach was implemented in Chrome at version 58.
But when I searched manually it seems like this feature exists from version 51 and should not be polyfilled in both cases.
https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach#browser_compatibility

How can I figure out why babel decided that the feature can be used only from version 58 and above.
Am I missing somthing here?

**Comment by fedeci** [comment]
#13082 seems to be a similar issue.

**Comment by nicolo-ribaudo** [comment, marked as answer]
The compatiblity data we use for `core-js` 3 are maintained by the `core-js` author at https://github.com/zloirock/core-js/tree/master/packages/core-js-compat. More specifically, I you can see at line 2996 of https://unpkg.com/browse/core-js-compat@3.10.1/data.json that `NodeList.forEach` is marked as supported on Chrome >= 58.

Usually MDN or Caniuse report a feature as supported even if it's only _partially_ supported (e.g. it doesn't work in every case), while `core-js` marks those cases as unsupported so that it can fix them.

If you want more info, I suggest asking directly in the `core-js` repository.
